go/apps/api/routes/v2_ratelimit_limit/handler.go

```diff
@@ -111,71 +131,34 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
 	}
 
 	// Verify permissions for rate limiting
-	err = h.Permissions.Check(
-		ctx,
-		auth.KeyID,
-		rbac.Or(
-			rbac.T(rbac.Tuple{
-				ResourceType: rbac.Ratelimit,
-				ResourceID:   namespace.ID,
-				Action:       rbac.Limit,
-			}),
-			rbac.T(rbac.Tuple{
-				ResourceType: rbac.Ratelimit,
-				ResourceID:   "*",
-				Action:       rbac.Limit,
-			}),
-		),
-	)
+	err = auth.Verify(ctx, keys.WithPermissions(rbac.Or(
+		rbac.T(rbac.Tuple{
+			ResourceType: rbac.Ratelimit,
+			ResourceID:   namespace.ID,
+			Action:       rbac.Limit,
+		}),
+		rbac.T(rbac.Tuple{
+			ResourceType: rbac.Ratelimit,
+			ResourceID:   "*",
+			Action:       rbac.Limit,
+		}),
+	)))
 	if err != nil {
 		return err
 	}
 
-	findOverrideMatchesArgs := db.ListRatelimitOverrideMatchesParams{
-		WorkspaceID: auth.AuthorizedWorkspaceID,
-		NamespaceID: namespace.ID,
-		Identifier:  req.Identifier,
-	}
-	ctx, overridesSpan := tracing.Start(ctx, "ListRatelimitOverrideMatches")
-	overrides, err := h.RatelimitOverrideMatchesCache.SWR(ctx, findOverrideMatchesArgs, func(ctx context.Context) ([]db.RatelimitOverride, error) {
-		return db.Query.ListRatelimitOverrideMatches(ctx, h.DB.RO(), findOverrideMatchesArgs)
-	}, func(err error) cache.Op {
-		if err == nil {
-			// everything went well and we have a namespace response
-			return cache.WriteValue
-		}
-		// this is a noop in the cache
-		return cache.Noop
-	})
-
-	overridesSpan.End()
-	if db.IsNotFound(err) {
-		return fault.New("namespace was deleted",
-			fault.Code(codes.Data.RatelimitNamespace.NotFound.URN()),
-			fault.Internal("namespace not found"), fault.Public("This namespace does not exist."),
-		)
-	}
-	if err != nil {
-		return err
-	}
 	// Determine limit and duration from override or request
 	var (
 		limit      = req.Limit
 		duration   = req.Duration
 		overrideId = ""
 	)
-	for _, override := range overrides {
-		if override.DeletedAtM.Valid {
-			continue
-		}
-		limit = int64(override.Limit)
-		duration = int64(override.Duration)
-		overrideId = override.ID
 
-		if override.Identifier == req.Identifier {
-			// Exact match takes precedence
-			break
-		}
+	found, override := matchOverride(req.Identifier, namespace)
```

Comment:
nit: typically in go the order of returns is [value, ok], mind if we change that for consistency?